src/tools/rbd_ggate/ggate_drv.c

```diff
@@ -0,0 +1,375 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```

Comment:
No. Although it is based on examples from FreeBSD source the code is mine. The reason I had to separate geom specific code to a C file is due to include geom/gate/g_gate.h, which further includes geom/geom.h. In the latest there is code like this:
```
LIST_ENTRY(g_class)     class;
```
which confuses c++ compiler.